fiboa_cli/datasets/pt.py

```diff
@@ -0,0 +1,84 @@
+from ..convert_utils import convert as convert_
+import geopandas as gpd
+import pandas as pd
+import fiona
+
+SOURCES = "https://www.ifap.pt/isip/ows/resources/2023/Continente.gpkg"
+#SOURCES = "https://www.ifap.pt/isip/ows/resources/2022/2022.zip"
+
+ID = "pt"
+TITLE = "Field boundaries for Portugal (identificação de parcelas)"
+DESCRIPTION = """Open field boundaries from Portugal"""
+PROVIDER_NAME = "IPAP - Instituto de Financiamento da Agricultura e Pescas"
+PROVIDER_URL = "https://www.ifap.pt/isip/ows/"
+ATTRIBUTION = None
+
+# Inspire license. Not 100% clear at source
+LICENSE = {"title": "No conditions apply", "href": "https://inspire.ec.europa.eu/metadata-codelist/ConditionsApplyingToAccessAndUse/noConditionsApply", "type": "text/html", "rel": "license"}
+
+COLUMNS = {
+    "geometry": "geometry",
+    "OSA_ID": "id",
+    "CUL_ID": "block_id",
+    "CUL_CODIGO": "crop_code",
+    "CT_português": "crop_name",
+    "Shape_Area": "area"
+}
+
+ADD_COLUMNS = {
+    "determination_datetime": "2023-01-01T00:00:00Z"
+}
+
+COLUMN_MIGRATIONS = {
+    "Shape_Area": lambda col: col / 10000.0
+}
+
+MISSING_SCHEMAS = {
+    "properties": {
+        "block_id": {
+            "type": "int64"
+        },
+        "crop_code": {
+            "type": "string"
+        },
+        "crop_name": {
+            "type": "string"
+        },
+    }
+}
+
+
+def file_migration(data, path, uri):
+    # filter for layers starting with Culturas_
+    gdfs = []
+    layers = fiona.listlayers(path)
+    for layer in layers:
+        if not layer.startswith("Culturas_"):
+            continue
+        data = gpd.read_file(path, layer=layer)
+        gdfs.append(data)
+
+    return pd.concat(gdfs)
```

Comment:
All data should already be loaded, so I think this needs to happen before to avoid wasting resources.
The file migration is more for merging different types of data, let me find a solution that is more integrated...

Reply:
Updated the CLI code to be able to filter by layers. Updated your template accordingly. Please check whether you are happy with it.

Reply:
Ok, I think the layer-filter is a useful feature. I have some doubts about a bug in the current implementation (handling of .parquet files): https://github.com/fiboa/cli/commit/49b54df29c7bbcf93e67d5668017c35d8c66bea9#r143467019

Currently, `_convert` is a complex function with many arguments. In a similar setup, I've used some kind of `Converter` super-class, overridable by all converters. This makes it easier to implement some custom unforeseen overrides. But for now this is workable.

Reply:
Good catch, thank you!
Fixed it in https://github.com/fiboa/cli/commit/f1140a60931baceaf1deee44322d235eb49bf0bd
Unfortunately, we don't have a GeoParquet source in converters anymore, so there's no test that catches this...

I agree that the convert function has grown significantly and should be refactored. Before I do this though, we should add more tests so that we can ensure a refactored version works properly. I generally started with testing various other commands today and I'll somewhat require tests for converters soon: #72 
Having that said, we should probably also test the converters a little more. Just relying on the converter in general, doesn't necessarily identifies "bugs" in the data.